cranelift-codegen/src/regalloc/liveness.rs

```diff
@@ -447,6 +447,19 @@ impl Liveness {
                     if let Some(constraint) = operand_constraints.next() {
                         lr.affinity.merge(constraint, &reginfo);
                     }
+
+                    let call_sig = func.dfg.call_signature(inst);
+                    if let Some(sig) = call_sig {
+                        if let crate::ir::InstructionData::CallIndirect { .. } = func.dfg[inst] {
```

Comment:
my great shame of this PR: the `CallIndirect` instruction takes one more argument than the number of arguments for the function to be called, which throws the affinity rewriting here all out of whack. Is there some nice way I couldn't find to generically talk about instruction arguments that line up with the call signature? I suppose worst case would be collecting signature params and zipping with instruction arguments in reverse, walking together, and assuming non-call-argument arguments will be at the start of the list.

Reply:
Figured out I should have been using `inst_variable_args` here and this is much nicer now :)